### PR TITLE
MINOR FIX: DataEquals function does not fail when data is different.

### LINKED ISSUE #9: DataEquals function does not fail when data is different.
For example in project Taarafo.Core there is PostServiceTests.cs in there is function SameValidationExceptionAs.

Lets take this test ShouldThrowValidationExceptionOnAddIfPostIsInvalidAndLogItAsync
In image below we see original test.
image

After deleting Data Setup tests still pass.
image

### PR DESCRIPTION
**Pull request by cjdutoit**
Closes #9 

Minor fix to ensure data dictionaries is compared LEFT to RIGHT and RIGHT to LEFT.
The comparison is currently only RIGHT to LEFT.

Also added a check to compare the count of the two dictionaries to return a FALSE if they don't match so that we get a faster output as the comparison would not be required then.

### CODE REVIEW

**Comment by cjdutoit** [comment]
@hassanhabib can you please review and merge?

File: Xeption.Tests/XeptionTests.Logic.cs [hunk 1]
```diff
@@ -295,5 +297,56 @@ public void ShouldTranscendDataFromAnotherExceptionWithCustomMessage()
...
+            // given
+            Dictionary<string, List<string>> randomDictionary =
+                CreateRandomDictionary();
+
+            var leftDictionary = randomDictionary;
+            var rightDictionary = randomDictionary.DeepClone();
+
+            rightDictionary.Add(
+                new MnemonicString().GetValue(),
+                new List<string> { new MnemonicString().GetValue() });
```
[the remark this answers is not in the document]
Comment (cjdutoit) [reply Xeption.Tests/XeptionTests.Logic.cs]: @ElbekDeveloper updated

File: Xeption.Tests/XeptionTests.Logic.cs [hunk 2]
```diff
@@ -295,5 +297,56 @@ public void ShouldTranscendDataFromAnotherExceptionWithCustomMessage()
...
+
+            var leftXeption = new Xeption();
+            leftXeption.AddData(leftDictionary);
```
[the remark this answers is not in the document]
Comment (cjdutoit) [reply Xeption.Tests/XeptionTests.Logic.cs:317]: @ElbekDeveloper updated

**Review by ElbekDeveloper — APPROVED** [review]

**Review by ElbekDeveloper — APPROVED** [review]

**Comment by cjdutoit** [comment]
Closing this one as #31 supersede this change

**Pull request closed without merging**